pkg/ebpf/tracer.go

```diff
@@ -7,16 +7,29 @@ import (
 	"fmt"
 	"io"
 	"io/fs"
+	"runtime"
 	"strings"
 	"time"
 
+	"github.com/cilium/ebpf"
 	"github.com/cilium/ebpf/ringbuf"
 	"github.com/netobserv/netobserv-ebpf-agent/pkg/flow"
 	"github.com/sirupsen/logrus"
 	"github.com/vishvananda/netlink"
 	"golang.org/x/sys/unix"
 )
 
+/*
+#include <time.h>
+static unsigned long long get_nsecs(void)
+{
+    struct timespec ts;
+    clock_gettime(CLOCK_MONOTONIC, &ts);
+    return (unsigned long long)ts.tv_sec * 1000000000UL + ts.tv_nsec;
+}
+*/
+import "C"
+
```

Comment:
That would force us to enable CGO (https://dave.cheney.net/2016/01/18/cgo-is-not-go).

Can't we just remove this and replace:
```go
timeNow := flow.Timestamp(C.get_nsecs())
```
by:
```go
timeNow := time.Now().UnixNano()
```
?

Reply:
The problem with the timestamp is that eBPF programs use CLOCK_MONOTONIC timestamp, however, the timeGo uses CLOCK_REALTIME. Hence I used the C wrapper.

`DEBU[0003] Monitor                                       component=ebpf.FlowTracer iface=ens6f0np0 timeNow=524694462887368 timeNowGo=1659419741836160042`

Above code snippet returns both time using C.get_nsecs() and time.Now().UnixNano(). Let me check if there is monotonic clock available in go.

Reply:
According to https://pkg.go.dev/time

> The Time returned by time.Now contains a monotonic clock reading.

Reply:
Ah, wait a minute, it's true that time.Now().UnixNano() is not monotonic.

The way of getting monotonic reads is by substracting two timestamps. e.g. subtract time.Now() to the agent start time:
https://pkg.go.dev/time#hdr-Monotonic_Clocks

```
start := time.Now()
... operation that takes 20 milliseconds ...
t := time.Now()
elapsed := t.Sub(start)
```

Reply:
As we had discussed, I have used monotime module. I used the package https://pkg.go.dev/github.com/gavv/monotime